kafka/producer/base.py

```diff
@@ -188,7 +211,12 @@ def _send_messages(self, topic, partition, *msg, **kwargs):
 
         if self.async:
             for m in msg:
-                self.queue.put((TopicAndPartition(topic, partition), m, key))
+                try:
+                    self.queue.put((TopicAndPartition(topic, partition), m, key), block=True,
+                                   timeout=self.batch_send_queue_max_wait)
+                except Full:
+                    log.exception('Queue full, failed to put message "{0}" in the async queue after {1} ms'.format(
+                        m, self.batch_send_queue_max_wait))
```

Comment:
Here are the behaviour expected:
If batch_send_queue_buffering_max_messages and batch_send_queue_max_wait are not set then the queue is an unlimited sized queue and put in the queue never wait or fail for Full exception.

If only batch_send_queue_max_wait is set, it won't change anything as the queue is unlimited in size.

If only batch_send_queue_buffering_max_messages is set to a value then the queue is limited in size. Put of messages will raise a Full exception if the queue is full as the default value of batch_send_queue_max_wait will induce no wait despite the parameter block set to True.

If both parameter are set the queue will be limited in size. Put of messages will wait the time set in batch_send_queue_max_wait (in second). If the queue is still full after this time an exception Full will be raised otherwise the put will succeed.

Let me know if you think that there is an issue on the code or if you think that this implementation is not appropriate.